onnxruntime/core/mlas/inc/mlas.h

```diff
@@ -91,7 +97,8 @@ MlasSgemm(
     size_t ldb,
     float beta,
     float* C,
-    size_t ldc
+    size_t ldc,
+    const ThreadPool* ExternalThreadPool
```

Comment:
Why does MLAS see a const thread pool? Seems weird for threading.cpp to then need to cast away the const-ness.

Reply:
It was originally to ensure that you understand you should not modify the ThreadPool.  Made the fix, but will think more on this.